### PR TITLE
Extract kroxylicious-api and kroxylicious-filter-api modules

### PR DESCRIPTION
**Pull request by robobario**
## Why

1. We want to create a clear separation of the interfaces and classes required to build custom filters and the internals of the proxy.
2. We want to re-use the Contributor interface for configuring and instantiating more services in future.

Filter authors should be able to create their own module that depends on `kroxylicious-filter-api`, implement a Filter using the interfaces provided, implement their own Contributor (or extend the BaseContributor) and include the service definition file in META-INF. Then their custom filters should be loaded into kroxylicious.

### CODE REVIEW

**Comment by robobario** [comment]
I've tested it's possible to extract the `MultiTenantTransformationFilter` into a module that only depends on kroxylicious-filter-api and implement a `Contributor` (plus add the service to META-INF) then see the integration tests pass once a dependency is added on that new module.

File: kroxylicious/pom.xml [hunk 1]
```diff
@@ -29,6 +29,11 @@
 
     <dependencies>
         <!-- We depend on the specific Netty dependencies not netty-all to reduce the size of fatjars -->
+        <dependency>
```
Comment (k-wall) [inline kroxylicious/pom.xml:32]: the netty comment has become separated from the netty dependencies.

**Review by k-wall — APPROVED** [review]

File: api/kroxylicious-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.kroxylicious</groupId>
+        <artifactId>kroxylicious-parent</artifactId>
+        <version>1.0-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>kroxylicious-api</artifactId>
```
Comment (SamBarker) [inline api/kroxylicious-api/pom.xml:21]: We should add a maven enforcer rule (in a follow up PR) to make sure the api packages don't depend on inappropriate things, such as netty.

**Review by SamBarker — APPROVED** [review]
LGTM

**Pull request merged**